File: core/src/main/java/hunternif/voxarch/plan/legacy/Prop.java
package hunternif.voxarch.plan.legacy;

import hunternif.voxarch.vector.Vec3;

/**
 * A fixed-size prop, i.e. a statue or a simple torch.
 * During planning it is effectively treated as a point.
 * @author Hunternif
 */
public class Prop extends Node {
	/** Defines what this prop actually is. */
	private final String name;

	/**
	 * The coordinates in blocks of the origin point relative to the origin
	 * of the parent room. Origin's location is different depending on mount type.
	 */
	public Prop(Room parent, String name, Vec3 origin, double rotationY) {
		super(parent, origin, rotationY);
		this.name = name;
	}
	
	public String getName() {
		return name;
	}
}

File: core/src/main/kotlin/hunternif/voxarch/plan/Prop.kt
package hunternif.voxarch.plan

import hunternif.voxarch.vector.Vec3

/**
 * ```
 * Y
 *  +----> X (East)
 *  |
 *  V
 *  Z
 * ```
 * A fixed-size prop, i.e. a statue or a simple torch.
 * During planning it is effectively treated as a point.
 * @param position will be copied
 */
open class Prop(position: Vec3, type: String) : Node(position) {
    init {
        this.tags += type
    }

    constructor(type: String): this(Vec3.ZERO, type)
}